### PR TITLE
Improve and fix tracking

### PR DESCRIPTION
**Pull request by fernandohds564**
- vectorialize multiparticle tracking;
 - reorder results for multiparticle tracking:
    - before:
   ```
    rin.shape = (Np, 6)
    rout.shape = (Np, 6, nr_turns)
    ```
    - now:
   ```
    rin.shape = (6, Np)
    rout.shape = (6, Np, nr_turns)
    ```
 - always `squeeze` results to have the minimum dimensions needed;
 - `ring_pass` now returns the positions of the particles at the beginning of every turn and the end of the last turn in case `turn_by_turn` is `True`.

Please, @murilobalves, @xresende @anacso17 and @AnaLuisaAR, test this version of the code with your scripts to catch any error I've let scape.

This code depends on lnls-fac/trackcpp#29. remember to change the branch of trackcpp and running:
`sudo make cleanall; make -j8; sudo make develop/install;`
before testing.

### CODE REVIEW

File: pyaccel/optics.py [hunk 1]
```diff
@@ -382,14 +384,13 @@ def calc_emittances(orbit, twiss, idx=0):
         acc, rin, nr_turns=nr_turns, turn_by_turn='closed', element_offset=0)
 
     if mode == 'fitting':
-        r = _np.dstack([rin[None, :, None], rout])
-        ax, bx = fitEllipse(r[0][0], r[0][1])
-        ay, by = fitEllipse(r[0][2], r[0][3])
+        ax, bx = fitEllipse(rout[0], rout[1])
```
Comment (xresende) [inline pyaccel/optics.py:387]: three-letter pylint rule in this section.

File: pyaccel/optics.py [hunk 2]
```diff
@@ -508,14 +510,13 @@ def get_mcf(accelerator, order=1, energy_offset=None):
         fp = _tracking.find_orbit4(accel, ene)
         X0 = _np.concatenate([fp.flatten(), [ene, 0]])
```
Comment (xresende) [inline pyaccel/optics.py:511]: here too.

File: pyaccel/optics.py (hunk 1 above)
Comment (fernandohds564) [reply pyaccel/optics.py:387]: this module has an infinity of pylint violations... :tired_face: 
I'll fix this one you've pointed out!

File: pyaccel/optics.py (hunk 2 above)
Comment (fernandohds564) [reply pyaccel/optics.py:511]: :tired_face:^2

File: pyaccel/optics.py (hunk 1 above)
Comment (xresende) [reply pyaccel/optics.py:387]: I know! I only commented the sections modified in this PR. later I'll create a _Standardize_ PRs for _pyaccel_ and _apsuite_ and submit them to you all.

Comment (fernandohds564) [reply pyaccel/optics.py]: done :face_with_head_bandage:

File: pyaccel/optics.py (hunk 2 above)
Comment (fernandohds564) [reply pyaccel/optics.py]: done :face_with_head_bandage:

**Pull request merged**